### PR TITLE
Improvment certificate api

### PR DESCRIPTION
**Pull request by ghost**
This PR improvments and fix the certificate api https://github.com/ILIAS-eLearning/ILIAS/pull/2132, which will be used in new my staff certificate view

- Fix filter by login not works (wrong variable check)
- Fix download link by using correct gui class and implement missing 'executeCommand' (Removed also the wrong ilCtrl comments which could crash the structure reload)

### CODE REVIEW

**Comment by mstuder** [comment]
Hi @mjansenDatabay 
We are in progress to integrate the certificate view for ILIAS Staff - . We are already using our proposal for the new tables for this purpose:
https://github.com/ILIAS-eLearning/ILIAS/pull/2186

So we use the api now for the first time in the real world. Can you please have a look at our optimization proposals?

File: Services/Certificate/classes/API/Repository/UserDataRepository.php [hunk 1]
```diff
@@ -129,41 +121,97 @@ public function getUserData(UserDataFilter $filter, array $ilCtrlStack) : array
...
-            case $filter->shouldSortByFirstNames():
-                $orderBy .= 'usr_data.firstname';
-                break;
+            $orderBy = ' ORDER BY ';
 
-            case $filter->shouldSortByLastNames():
-                $orderBy .= 'usr_data.lastname';
-                break;
+            foreach ($sorts as [$key, $reverse]) {
+                $reverse = $reverse ? ' DESC ' : ' ASC ';
```
Comment (mjansenDatabay) [inline Services/Certificate/classes/API/Repository/UserDataRepository.php]: Please rename this variable to `$direction` or something similar.

File: Services/Certificate/classes/API/Download/class.ilUserCertificateApiGUI.php [hunk 2]
```diff
@@ -46,10 +52,34 @@ public function __construct(
...
+
 
         $this->language->loadLanguageModule('cert');
     }
 
+
+    /**
+     *
+     */
+    public function executeCommand() : void
```
Comment (mjansenDatabay) [inline Services/Certificate/classes/API/Download/class.ilUserCertificateApiGUI.php:68]: Thanks a lot. This was indeed missing. Niels tested this by calling `download` directly in the consumer, so the absence was not really obvious.

File: Services/Certificate/classes/API/Repository/UserDataRepository.php [hunk 3]
```diff
@@ -129,41 +121,97 @@ public function getUserData(UserDataFilter $filter, array $ilCtrlStack) : array
...
     private function createOrderByClause(UserDataFilter $filter) : string
     {
-        $orderBy = ' ORDER BY ';
+        $sorts = $filter->getSorts();
 
-        switch (true) {
-            case $filter->shouldSortByLogins():
-                $orderBy .= 'usr_data.login';
-                break;
+        if (!empty($sorts)) {
```
Comment (mjansenDatabay) [inline Services/Certificate/classes/API/Repository/UserDataRepository.php]: I suggest an `early return` here, if `$sorts` is empty. With this we could get rit of the `else` case and one indentation level for the main logic.

File: Services/Certificate/classes/API/Repository/UserDataRepository.php [hunk 4]
```diff
@@ -186,7 +234,7 @@ private function createWhereCondition(UserDataFilter $filter) : string
         }
 
         $login = $filter->getUserLogin();
-        if (null !== $lastName) {
+        if (null !== $login) {
```
Comment (mjansenDatabay) [inline Services/Certificate/classes/API/Repository/UserDataRepository.php:189]: Thx!

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]
@sragfw / @mstuder  : Thanks a lot for this improvements. I like the new `multi sort` capability and only had some minor remarks.

But: Could you please create a separate PR for the limit/offset topic? I am not sure if this is beneficial for the consumer/client, so I would like to decide/discuss upon this separately.

**Comment by ghost** [comment]
Yes, I create a second PR https://github.com/ILIAS-eLearning/ILIAS/pull/2216

File: Services/Certificate/classes/API/Repository/UserDataRepository.php (hunk 1 above)
Comment (ghost) [reply Services/Certificate/classes/API/Repository/UserDataRepository.php]: Made in https://github.com/ILIAS-eLearning/ILIAS/pull/2216

File: Services/Certificate/classes/API/Repository/UserDataRepository.php (hunk 3 above)
Comment (ghost) [reply Services/Certificate/classes/API/Repository/UserDataRepository.php]: Made in https://github.com/ILIAS-eLearning/ILIAS/pull/2216

**Review by mjansenDatabay — APPROVED** [review]

**Pull request merged**